### ISSUE TITLE
Review & fix neumorphic styles

### ISSUE THREAD

**Issue by Xiaoy312** [unoplatform/Uno.Gallery#931]
For each Neumorphic styles
- going over figma design for dark/light, to see if a pattern is present to extract reusable color/shadows
- ensure its ShadowContainer's child doesnt have a background, and is placed on the SC itself (including VisualState.Setter target as well).
- ensure property surface color, and shadow values are used for every visual state.

**Comment by Xiaoy312** [comment]
fixme:
- !~ switching shadows between these that generates different canvas layout can cause shadow to be drawn at incorect location (see checkbox): 
  ![image](https://github.com/unoplatform/Uno.Gallery/assets/2359550/4cb466d7-fbeb-4985-a058-07a7f5ab8428)

  > workaround available: transparent shadow to align various states' layout
- painted shadow is slightly off in some case:
  ![image](https://github.com/unoplatform/Uno.Gallery/assets/2359550/a90adaa8-365f-4e1b-ab7f-840b2725e172) (chip)
- !! \#NeumorphicControlCombo selecting "TabBar" and then selecting any other item cause the app to crash:
  > System.NullReferenceException    HResult=0x80004003    Message=Object reference not set to an instance of an object.    Source=Uno.UI    StackTrace:     at Microsoft.UI.Xaml.Controls.MenuFlyout.OnClosed() in /__w/1/s/src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyout.cs:line 193  
- !!! with the current impl it is not possible to insert border between background and inset-shadow: (drop-shadow, background, border, inset-shadow)

todo:
- combobox: missing flyout design
- textbox,passwordbox: improper cursor placement (we should have a branch in toolkit with the proper setup)

**Comment by agneszitte** [comment]
> fixme:
> 
> * !~ switching shadows between these that generates different canvas layout can cause shadow to be drawn at incorect location (see checkbox):
>   ![image](https://user-images.githubusercontent.com/2359550/264550161-4cb466d7-fbeb-4985-a058-07a7f5ab8428.png)
>   > workaround available: transparent shadow to align various states' layout

I have opened an issue for this one -> https://github.com/unoplatform/uno.toolkit.ui/issues/794

**Comment by Xiaoy312** [comment]
they all need separate issues for tracking, 
i just dont have time to delve into them yet or to gather info to fill the issue
so i just listed them here

#\2 idk if related for neumorphic or not, that is what im observing

**Comment by agneszitte** [comment]
> they all need separate issues for tracking, i just dont have time to delve into them yet or to gather info to fill the issue so i just listed them here
> 
> #\2 idk if related for neumorphic or not, that is what im observing

thanks for the feedback, I will help you open proper issues for them
Regarding the platforms, where do you see them? do you have this info, please?

**Comment by agneszitte** [comment]
Here are the related I have opened

1. 
	> * painted shadow is slightly off in some case:
	>   ![image](https://user-images.githubusercontent.com/2359550/264549934-a90adaa8-365f-4e1b-ab7f-840b2725e172.png) (chip)
	
## https://github.com/unoplatform/Uno.Gallery/issues/938

2. 
	> * !! #NeumorphicControlCombo selecting "TabBar" and then selecting any other item cause the app to crash:
	>   > System.NullReferenceException    HResult=0x80004003    Message=Object reference not set to an instance of an object.    Source=Uno.UI    StackTrace:     at Microsoft.UI.Xaml.Controls.MenuFlyout.OnClosed() in /__w/1/s/src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyout.cs:line 193
	
## https://github.com/unoplatform/Uno.Gallery/issues/939

3.
	> * !!! with the current impl it is not possible to insert border between background and inset-shadow: (drop-shadow, background, border, inset-shadow)
	
## https://github.com/unoplatform/uno.toolkit.ui/issues/795

4.
	> * combobox: missing flyout design

	Neumorphic style for flyout is a "Nice to have" at this moment, so not a priority for today
	@eriklimakc already did some changes for this -> https://github.com/unoplatform/Uno.Gallery/pull/905
	But it was blocked by -> https://github.com/unoplatform/uno.toolkit.ui/issues/760

5.
	> * textbox,passwordbox: improper cursor placement (we should have a branch in toolkit with the proper setup)
	
	I will test this one before I decide to open an issue for it

**Comment by agneszitte** [comment]
6.
	> * textbox,passwordbox: improper cursor placement (we should have a branch in toolkit with the proper setup)
	
Yes seems a Skia-only issue and can also be reproduced on the Fluent TextBox also

**Comment by agneszitte** [comment]
Re-Open as work ongoing for #944

**Comment by agneszitte** [comment]
Re-Open for backport PR

**Comment by agneszitte** [comment]
@eriklimakc is currently working on the last fixes for ToggleSwitch

**Comment by agneszitte** [comment]
Re-Open for backport PR
